airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/pipeline.py

```diff
@@ -109,27 +104,21 @@ async def run_connectors_pipelines(
         else:
             dockerd_service = docker.with_global_dockerd_service(dagger_client)
 
-        async with anyio.create_task_group() as tg_main:
-            tg_main.start_soon(dockerd_service.sync)
-            await (  # Wait for the docker service to be ready
-                dagger_client.container()
-                .from_(DOCKER_CLI_IMAGE)
-                .with_env_variable("DOCKER_HOST", f"tcp://{DOCKER_HOST_NAME}:{DOCKER_HOST_PORT}")
-                .with_service_binding(DOCKER_HOST_NAME, dockerd_service)
-                .with_exec(["docker", "info"])
-            )
-            async with anyio.create_task_group() as tg_connectors:
-                for context in contexts:
-                    context.dagger_client = dagger_client.pipeline(f"{pipeline_name} - {context.connector.technical_name}")
-                    context.dockerd_service = dockerd_service
-                    tg_connectors.start_soon(
-                        connector_pipeline,
-                        context,
-                        CONNECTOR_LANGUAGE_TO_FORCED_CONCURRENCY_MAPPING.get(context.connector.language, default_connectors_semaphore),
-                        *args,
-                    )
-            # When the connectors pipelines are done, we can stop the dockerd service
-            tg_main.cancel_scope.cancel()
+        await dockerd_service.start()
```

Comment:
💅 nit: Should/could we turn this to an async with

```py
async with dockerd_service.start():
```

Reply:
It'd be nice but I don't think we can as it's not returning a context manager